platforms/axplat-aarch64-peripherals/src/gic.rs

```diff
@@ -1,25 +1,34 @@
 //! ARM Generic Interrupt Controller (GIC).
 
-use arm_gicv2::{GicCpuInterface, GicDistributor};
-use axplat::irq::{HandlerTable, IrqHandler};
+use arm_gic_driver::v2::*;
+use axplat::irq::{HandlerTable, IpiTarget, IrqHandler};
 use axplat::mem::VirtAddr;
 use kspin::SpinNoIrq;
 use lazyinit::LazyInit;
+use spin::{Mutex, Once};
 
 /// The maximum number of IRQs.
 const MAX_IRQ_COUNT: usize = 1024;
 
-static GICD: LazyInit<SpinNoIrq<GicDistributor>> = LazyInit::new();
+static GIC: Mutex<Gic> = Mutex::new(unsafe {
+    Gic::new(
+        arm_gic_driver::VirtAddr::new(0),
+        arm_gic_driver::VirtAddr::new(0),
+        None,
+    )
+});
 
-// per-CPU, no lock
-static GICC: LazyInit<GicCpuInterface> = LazyInit::new();
+static TRAP_OP: Once<TrapOp> = Once::new();
```

Comment:
No need to make `TrapOp` a global value, call `trap_operations()` in `handle_irq` directly.

Reply:
Yes, you can also call `trap_operations` in `init_gicc`.